Objects/typeobject.c

```diff
@@ -5960,45 +5944,16 @@ slot_sq_length(PyObject *self)
     return len;
 }
 
-/* Super-optimized version of slot_sq_item.
-   Other slots could do the same... */
 static PyObject *
 slot_sq_item(PyObject *self, Py_ssize_t i)
 {
-    PyObject *func, *ival = NULL, *retval = NULL;
-    descrgetfunc f;
-
-    func = _PyType_LookupId(Py_TYPE(self), &PyId___getitem__);
-    if (func == NULL) {
-        PyObject *getitem_str = _PyUnicode_FromId(&PyId___getitem__);
-        PyErr_SetObject(PyExc_AttributeError, getitem_str);
+    PyObject *retval;
+    PyObject *ival = PyLong_FromSsize_t(i);
+    if (ival == NULL)
         return NULL;
-    }
-
-    f = Py_TYPE(func)->tp_descr_get;
-    if (f == NULL) {
-        Py_INCREF(func);
-    }
-    else {
-        func = f(func, self, (PyObject *)(Py_TYPE(self)));
-        if (func == NULL) {
-            return NULL;
-        }
-    }
-
-    ival = PyLong_FromSsize_t(i);
-    if (ival == NULL) {
-        goto error;
-    }
-
-    retval = PyObject_CallFunctionObjArgs(func, ival, NULL);
-    Py_DECREF(func);
+    retval = call_method(self, &PyId___getitem__, &ival, 1);
```

Comment:
This change optimizes the `__getitem__` slot if I understand correctly, since it handles bound/unbound methods. Good :-) It might be useful to mention it.

About &ival: FYI I removed the _PyObject_CallArg1() macro since it increased the stack usage: https://bugs.python.org/issue28858